src/result_processor.c

```diff
@@ -140,6 +138,7 @@ int baseResultProcessor_Next(ResultProcessorCtx *ctx, SearchResult *res) {
   res->indexResult = r;  // q->opts.needIndexResult ? r : NULL;
 
   res->score = 0;
+  ++dmd->ref_count;  // we increase the refcount so the dmd will not be freed till we done with it
```

Comment:
Use DMD_Incref/DMD_Decref rather than direct access. Incref should be a macro, while decref should be something like

```c
static inline DMD_Decref(RSDocumentMetadata *md) {
    if (!--md->ref_count) { DMD_Free(md); }
}
```